### ISSUE TITLE
[Bug]: dataNode failed to SaveBinlogPaths

### ISSUE THREAD

**Issue by xiaozhuang-a** [milvus-io/milvus#26005]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: v2.2.10
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

datanode  Continuously reporting errors and restarting
Similar to this issue https://github.com/milvus-io/milvus/issues/23240

```bash

[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flow_graph_insert_buffer_node.go:469] ["insertBufferNode syncing BufferData"] [segmentID=442776935130016392] [flushed=false] [dropped=false] [auto=false] [position="channel_name:\"by-dev-rootcoord-dml_444_442776935130016343v0\" msgID:\"\\010\\225\\227\\202\\001\\020\\214\\230\\001\\030\\000 \\000\" msgGroup:\"by-dev-dataNode-55-by-dev-rootcoord-dml_444_442776935130016343v0\" timestamp:443192549629755393 "] [channel=by-dev-rootcoord-dml_444_442776935130016343v0]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flush_manager.go:287] ["handling insert task"] ["segment ID"=442776935130016392] [flushed=false] [dropped=false] [position="channel_name:\"by-dev-rootcoord-dml_444_442776935130016343v0\" msgID:\"\\010\\225\\227\\202\\001\\020\\214\\230\\001\\030\\000 \\000\" msgGroup:\"by-dev-dataNode-55-by-dev-rootcoord-dml_444_442776935130016343v0\" timestamp:443192549629755393 "]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flush_manager.go:139] ["new flush task runner created and initialized"] ["segment ID"=442776935130016392] ["pos message ID"="\u0008\ufffd\ufffd\ufffd\u0001\u0010\ufffd\ufffd\u0001\u0018\u0000 \u0000"]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flush_task.go:134] ["running flush insert task"] ["segment ID"=442776935130016392] [flushed=false] [dropped=false] [position="channel_name:\"by-dev-rootcoord-dml_444_442776935130016343v0\" msgID:\"\\010\\225\\227\\202\\001\\020\\214\\230\\001\\030\\000 \\000\" msgGroup:\"by-dev-dataNode-55-by-dev-rootcoord-dml_444_442776935130016343v0\" timestamp:443192549629755393 "] [PosTime=2023/07/29 15:43:27.218 +00:00]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/channel_meta.go:453] ["roll pk stats"] ["segment id"=442776935130016392]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flow_graph_delete_node.go:118] ["DeleteNode receives flush message"] [segIDs="[442776935130016392]"] [vChannelName=by-dev-rootcoord-dml_444_442776935130016343v0] [posTime=2023/07/29 15:43:27.218 +00:00]
[2023/07/29 15:43:27.374 +00:00] [INFO] [datanode/flush_manager.go:314] ["handling delete task"] ["segment ID"=442776935130016392]
[2023/07/29 15:43:27.451 +00:00] [WARN] [datanode/flush_manager.go:853] ["failed to SaveBinlogPaths"] ["segment ID"=442776935130016392] [error="All attempts results:\nattempt #1:segment 442776935130016392 not found\nattempt #2:segment 442776935130016392 not found\nattempt #3:segment 442776935130016392 not found\nattempt #4:segment 442776935130016392 not found\nattempt #5:segment 442776935130016392 not found\nattempt #6:segment 442776935130016392 not found\nattempt #7:segment 442776935130016392 not found\nattempt #8:segment 442776935130016392 not found\nattempt #9:segment 442776935130016392 not found\nattempt #10:segment 442776935130016392 not found\nattempt #11:segment 442776935130016392 not found\n"]
panic: All attempts results:
attempt #1:segment 442776935130016392 not found
attempt #2:segment 442776935130016392 not found
attempt #3:segment 442776935130016392 not found
attempt #4:segment 442776935130016392 not found
attempt #5:segment 442776935130016392 not found
attempt #6:segment 442776935130016392 not found
attempt #7:segment 442776935130016392 not found
attempt #8:segment 442776935130016392 not found
attempt #9:segment 442776935130016392 not found
attempt #10:segment 442776935130016392 not found
attempt #11:segment 442776935130016392 not found


goroutine 6092910 [running]:
github.com/milvus-io/milvus/internal/datanode.flushNotifyFunc.func1(0xc09a773ae0)
   /go/src/github.com/milvus-io/milvus/internal/datanode/flush_manager.go:857 +0x13f6
github.com/milvus-io/milvus/internal/datanode.(*flushTaskRunner).waitFinish(0xc0c102e180, 0xc011be0a80, 0xc0a69764d0)
   /go/src/github.com/milvus-io/milvus/internal/datanode/flush_task.go:204 +0xdb
created by github.com/milvus-io/milvus/internal/datanode.(*flushTaskRunner).init.func1
   /go/src/github.com/milvus-io/milvus/internal/datanode/flush_task.go:121 +0xf1

```

### Steps To Reproduce

```markdown
unknown
```

### Milvus Log

https://github.com/xiaozhuang-a/demo/blob/main/datanode_0730.log.zip

**Comment by xiaozhuang-a** [comment]
Could you please provide a response and solution as soon as possible，ths

**Comment by aoiasd** [comment]
Could you provide the log of datacoord, this bug was cause by datacoord not know this segment or has removed it.
And this question my has fixed at 2.2.12.

**Comment by xiaofan-luan** [comment]
we would like t work on it
